independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java

```diff
@@ -30,8 +32,9 @@
 
     private static final String PROJECT_BASEDIR = "${project.basedir}";
     private static final String POM_XML = "pom.xml";
-    private static final String REVISION = "revision";
-    static final String REVISION_EXPR = "${revision}";
+
+    private static final Pattern UNRESOLVED_VERSION_PATTERN = Pattern
+            .compile(Pattern.quote("${") + "(revision|sha1|changelist)" + Pattern.quote("}"));
```

Comment:
I believe you can use other property names as the version. Shouldn't we just test if the version matches `\$\{.*\}`?

Reply:
That's what I first thought, too. But that is _not_ the case as @aloubyansky pointed out here: https://github.com/quarkusio/quarkus/pull/7786#discussion_r391700587

See also the following bit from https://maven.apache.org/maven-ci-friendly.html:
![grafik](https://user-images.githubusercontent.com/22860528/77228629-e0d4fb00-6b88-11ea-9147-2035e4dc2a4e.png) 

In fact I had a very strange problem when I used a custom property in another project and back then I did not know that custom properties are not allowed.

Reply:
Interesting. Didn't know about that

Reply:
Yeah, me neither. I should have added a small comment to this field but I forgot about that...

Reply:
There is still time, the PR wasn't merged yet :wink:

Reply:
Ok, will add a comment once the last check has finished (Windows JDK 11 JVM Tests).